changes/2020-06-09_wrapping-key-identifiers/change.md

```diff
@@ -0,0 +1,145 @@
+Follow-up changes that define
+whether implementations expose these concepts
+MAY affect implementations.
+
+## Definitions
+
+### Conventions used in this document
+
+The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in
+[RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Summary
+
+We have referenced several terms throughout various specification documents
+that are never defined:
+"key namespace",
+"key name",
+"provider ID",
+and "provider info".
+This change defines each of these terms
+and describes their relationships to each other
+and the keyrings they represent.
+
+## Out of Scope
+
+- Whether, or how, each keyring exposes any of these values at runtime is out of scope.
+
+- Specific values for these terms for any specific keyring is out of scope.
+
+## Motivation
+
+Various specification documents reference
+"key namespace",
+"key name",
+"provider ID",
+and "provider info",
+but we never define what these terms mean
+or how they relate to each other.
+
+In order to ensure the specification documents accurately describe keyring behavior,
+we need to define all terms that they use.
+
+## Drawbacks
+
+This change SHOULD NOT introduce any drawbacks.
+It is identifying and describing
+terms that already exist
+and the existing relationship between them.
+
+## Security Implications
+
+This change SHOULD NOT have any security implications.
+
+## Operational Implications
+
+This change SHOULD NOT have any operational implications.
+
+## Guide-level Explanation
+
+Key namespace and key name are configuration values that MAY determine the behavior of a keyring.
+
+Provider ID and provider info are values that identify the keyring configuration
+that can fulfill the keyring's decryption contract.
+The keyring attaches these values to the encrypted data key ciphertext.
+
+## Reference-level Explanation
+
+"Key namespace", "key name", "provider ID", "provider info"
+are all concepts that identify a wrapping key.
+
+### key namespace
+
+A configuration value for a keyring
+that identifies the grouping or categorization
+for the wrapping keys that keyring can access.
+
+### key name
+
+A configuration value for a keyring
+that identifies a single wrapping key
+within a key namespace.
+
+### provider ID
+
+An output value returned by a keyring on encrypt
+as part of an encrypted data key structure
+that identifies the grouping or categorization
+for the keyring that can fulfill this keyring's decryption contract.
+
+The provider ID SHOULD be equal to the key namespace.
+
+### provider info
+
+An output value returned by a keyring on encrypt
+as part of an encrypted data key structure
+that provides necessary information for a keyring
+to fulfill this keyring's decryption contract.
+
+The provider info SHOULD be equal to the key name.
+
+One example of a keyring where the key name and provider info can differ
```

Comment:
It the Raw AES keyring another good example?

Reply:
It's an example...I don't know that it's a *good* example. :/

I'd rather not call attention to it shoving extra stuff into the provider info because that's a pattern that we do not want to encourage in the future.

Reply:
totally fair